### ISSUE TITLE
Support parameter optimization for alpha backtests

### ISSUE THREAD

**Issue by knoomdevbot** [knoomdevbot/quant-for-agent#62]
## Feature request

Add a first-class qfa parameter optimization / parameter sweep workflow for alpha backtests.

## Motivation

Many alpha models have tunable parameters: lookback windows, thresholds, holding periods, rebalance cadence, volatility targets, sleeve caps, stop/exit rules, z-score cutoffs, event lags, etc. In current AOI research we repeatedly need to run the same alpha over a bounded parameter grid to avoid judging a hypothesis from one arbitrary setting. Doing this outside qfa creates duplicated evaluator scripts and inconsistent artifact schemas.

A qfa-native optimizer would improve research velocity and reproducibility by letting agents ask:

> For this alpha, over this universe/date range/bar unit, which parameter region is robust after costs and controls?

This should be research tooling, not a curve-fit guarantee. The output should emphasize best, median/typical, robustness/fragility, and holdout/walk-forward diagnostics rather than only the single best in-sample setting.

## Proposed CLI shape

Possible command:

```bash
qfa backtest optimize path/to/alpha.py \
  --symbols SPY,QQQ,IWM,TLT,GLD \
  --start 2021-01-01 \
  --end 2026-01-01 \
  --timeframe 15min \
  --initial-cash 100000 \
  --param-grid params.yaml \
  --objective sharpe \
  --cost-bps 10 \
  --random-windows 50 \
  --walk-forward \
  --db /tmp/qfa-opt.sqlite3
```

Alternative inline form:

```bash
qfa backtest optimize model.py \
  --symbols SPY,QQQ \
  --start 2021-01-01 \
  --end 2026-01-01 \
  --param lookback=20,40,60,120 \
  --param threshold=0.5,1.0,1.5 \
  --param hold_days=1,3,5
```

## Model parameter interface

Support at least one clear way for qfa to pass parameters into an alpha model, e.g.:

1. Add `context.parameters` / `context.params` for `generate_signals(context)`.
2. Allow model modules to expose a parameter schema, e.g.:

```python
PARAMETERS = {
    "lookback": [20, 60, 120],
    "threshold": [0.5, 1.0, 1.5],
    "hold_days": [1, 3, 5],
}
```

3. Allow external YAML/JSON grid files so the same model can be swept without editing code.

Example `params.yaml`:

```yaml
parameters:
  lookback: [20, 40, 80, 120]
  threshold: [0.5, 1.0, 1.5]
  hold_days: [1, 3, 5]
constraints:
  max_combinations: 200
  skip_if:
    - hold_days > lookback
```

## Optimization modes

Minimum useful modes:

- grid search over bounded explicit parameter lists;
- optional random search for large grids;
- train/test split or walk-forward evaluation;
- random-period window evaluation for each candidate parameter set;
- top-N candidate retention plus median/typical result reporting;
- optional parallel execution with deterministic seeds.

## Output requirements

The optimizer should emit machine-readable JSON and persist results to qfa's DB. Include:

- model path and model hash/version if available;
- symbols/universe/date range/timeframe/base bar unit;
- data source and asset class;
- parameter grid definition and number of combinations evaluated/skipped;
- objective metric and tie-break rules;
- cost/slippage assumptions;
- best parameter set by objective;
- median/typical parameter-set metrics;
- top-N parameter sets;
- per-parameter robustness/fragility diagnostics;
- train/test or walk-forward metrics when requested;
- random-window distribution for best and representative parameter sets;
- warnings for overfitting risk, sparse trades, low activation, high turnover, or high sensitivity.

Suggested JSON shape:

```json
{
  "id": "...",
  "model_path": "model.py",
  "symbols": ["SPY", "QQQ"],
  "start": "2021-01-01",
  "end": "2026-01-01",
  "timeframe": "15min",
  "base_bar_unit": "15min",
  "objective": "sharpe_after_costs",
  "costs_bps": 10,
  "grid": {"lookback": [20, 60, 120], "threshold": [0.5, 1.0]},
  "evaluated_combinations": 6,
  "best_parameters": {"lookback": 60, "threshold": 1.0},
  "best_metrics": {"sharpe": 1.2, "total_return": 0.42, "max_drawdown": -0.12, "turnover": 2.1},
  "median_metrics": {"sharpe": 0.4},
  "top_results": [],
  "walk_forward": {},
  "random_windows": {},
  "warnings": []
}
```

## Guardrails / anti-overfit requirements

This feature should not encourage blindly picking the best backtest. Please include guardrails:

- require bounded grids or explicit max-combination limits;
- report best and median/typical metrics, not just best;
- support out-of-sample / walk-forward splits;
- support random-window robustness;
- flag parameters that win only in one period or have poor recent/holdout behavior;
- flag very high turnover / cost sensitivity;
- preserve all tested parameter metadata for reproducibility;
- avoid retaining raw bars/equity curves by default unless explicitly requested.

## Interaction with AOI alpha research

AOI research currently requires:

- real market data only; no CSV-backed qfa research;
- exactly 10 bps one-way canonical cost for grading unless explicitly overridden;
- bar-unit comparison across 15min / 1hour / 1day when available;
- bounded, mechanism-plausible parameter sweeps before terminal pruning;
- last-5y / last-1y and rolling 3y diagnostics;
- compact artifacts suitable for GitHub review.

A native qfa optimizer would let those research contracts be implemented consistently without each alpha agent writing its own transient sweep runner.

## Acceptance criteria

- [ ] `qfa backtest optimize` or equivalent command exists.
- [ ] Parameters can be passed to models through a documented interface such as `context.parameters`.
- [ ] Grid file and/or repeated `--param name=value1,value2` syntax is supported.
- [ ] Optimizer runs real-data backtests without `--data-csv` for research workflows.
- [ ] Results are persisted and viewable via qfa CLI/DB.
- [ ] JSON output includes best, median/typical, top-N, and robustness diagnostics.
- [ ] Walk-forward or train/test mode exists.
- [ ] Random-window robustness can be requested.
- [ ] Tests cover model parameter injection, grid parsing, skipped/invalid parameter combos, result persistence, and deterministic output ordering.
